juriscraper/opinions/united_states/federal_special/tax.py

```diff
@@ -30,6 +31,22 @@ def __init__(self, *args, **kwargs):
         }
 
     def _download(self, request_dict={}):
+        """Download from api
+
+        The tax court switches between blue and green deploys so we need to
+        check which one is current before we continue
+
+        :param request_dict:
+        :return: None
+        """
+        if not self.set_blue_green and not self.test_mode_enabled():
+            check = self.request["session"].get(self.url)
+            if check.status_code != 200:
```

Comment:
Every get request must have a timeout. If we forget these, the code can hang, literally forever, which would stop our scraper for all courts. It's a massive wart in the requests library that they're incapable of fixing.

Reply:
I assume it's subclassed with a timeout.

Reply:
Oh, maybe you're right. I'm pretty paranoid about it.

Reply:
```
    def _request_url_get(self, url):
        """Execute GET request and assign appropriate request dictionary
        values
        """
        self.request["url"] = url
        self.request["response"] = self.request["session"].get(
            url,
            headers=self.request["headers"],
            verify=self.request["verify"],
            timeout=60,
            **self.request["parameters"],
        )

    def _request_url_post(self, url):
        """Execute POST request and assign appropriate request dictionary values"""
        self.request["url"] = url
        self.request["response"] = self.request["session"].post(
            url,
            headers=self.request["headers"],
            verify=self.request["verify"],
            data=self.parameters,
            timeout=60,
            **self.request["parameters"],
        )
```

Reply:
Just to square the circle, I believe you were already paranoid enough about timeouts within juriscraper